packages/task/src/browser/task-service.ts

```diff
@@ -210,10 +210,52 @@ export class TaskService implements TaskConfigurationClient {
 
         // open terminal widget if the task is based on a terminal process (type: shell)
         if (taskInfo.terminalId !== undefined) {
-            this.attach(taskInfo.terminalId, taskInfo.taskId);
```

Comment:
Would you consider to leave a widget creation in the `attach` method to avoid duplication of a big code fragment?
`TaskInfo` can be passed into the `attach` method in order to get all the necessary information.